templates/partials/summary/summary.html

```diff
@@ -3,44 +3,55 @@
 
-  {%- for group in content.summary.groups if group.blocks -%}
-    {% do summary_groups.append
+
+  {%- for section in content.summary.sections -%}
+    {% set summary_groups = [] %}
+    {%- if section.groups -%}
+      {%- for group in section.groups -%}
+        {%- if group.blocks -%}
+          {% do summary_groups.append
+            (
+              {
+                  "groupTitle": group.title if group.title else None,
+                  "id": group.id if group.id else None,
+                  "rows": map_summary_item_config(
+                    group=group,
+                    summary_type=summary_type,
+                    answers_are_editable=content.summary.answers_are_editable,
+                    no_answer_provided=_("No answer provided"),
+                    remove_link_text=_("Remove") if not view_submitted_response else "",
+                    remove_link_aria_label=_("Remove your answer for:") if not view_submitted_response else "",
+                    edit_link_text=_("Change") if not view_submitted_response else "",
+                    edit_link_aria_label=_("Change your answer for:") if not view_submitted_response else "",
+                    calculated_question=content.summary.calculated_question,
+                    icon=None
+                  ),
+                  "classes": "ons-u-mt-m" if loop.index > 1 else "",
+                  "placeholderText": group.placeholder_text,
+                  "summaryLink": group.links.add_link,
+                }
+            )
+          %}
+        {%- endif -%}
+      {%- endfor -%}
+    {%- endif -%}
+    {% do summary_sections.append
       (
         {
-          "summaryTitle": group.section_title,
-          "groups": [
-            {
-              "groupTitle": group.title if group.title else None,
-              "id": group.id if group.id else None,
-              "rows": map_summary_item_config(
-                group=group,
-                summary_type=summary_types,
-                answers_are_editable=content.summary.answers_are_editable,
-                no_answer_provided=_("No answer provided"),
-                remove_link_text=_("Remove") if not view_submitted_response else "",
-                remove_link_aria_label=_("Remove your answer for:") if not view_submitted_response else "",
-                edit_link_text=_("Change") if not view_submitted_response else "",
-                edit_link_aria_label=_("Change your answer for:") if not view_submitted_response else "",
-                calculated_question=content.summary.calculated_question,
-                icon=None
-              ),
-              "classes": "ons-u-mt-m" if loop.index > 1 else "",
-              "placeholderText": group.placeholder_text,
-              "summaryLink": group.links.add_link,
-            }
-          ]
+          "summaryTitle": section.title if summary_type == "Summary" and summary_groups else None,
+          "groups": summary_groups
```

Comment:
I don't get the purpose of this?. Ends up with empty sections?.

![image](https://github.com/ONSdigital/eq-questionnaire-runner/assets/35296336/74398094-2893-404a-9a64-9d2422b4633d)

Reply:
The if is currently guarding against the sections with empty groups and blocks that are being generated when there is only a list collector.

Reply:
I've add logic to filter these empty ones now 👍